src/WebJobs.Script/StorageProvider/StorageClientProvider.cs

```diff
@@ -96,27 +95,29 @@ public virtual bool TryGetFromConnectionString(string connectionString, out TCli
         /// </summary>
         /// <param name="name">Name of the connection to use</param>
         /// <param name="resolver">A resolver to interpret the provided connection <paramref name="name"/>.</param>
+        /// <param name="configuration">Configuration to use</param>
         /// <returns>Client that was created.</returns>
-        public virtual TClient Get(string name, INameResolver resolver)
+        public virtual TClient Get(string name, INameResolver resolver, IConfiguration configuration)
         {
             var resolvedName = resolver.ResolveWholeString(name);
-            return this.Get(resolvedName);
+            return this.Get(resolvedName, configuration);
         }
 
         /// <summary>
         /// Gets the storage client specified by <paramref name="name"/>
         /// </summary>
         /// <param name="name">Name of the connection to use</param>
+        /// <param name="configuration">Configuration to use</param>
         /// <returns>Client that was created.</returns>
-        public virtual TClient Get(string name)
+        public virtual TClient Get(string name, IConfiguration configuration)
         {
             if (string.IsNullOrWhiteSpace(name))
             {
                 name = ConnectionStringNames.Storage; // default
             }
 
             // $$$ Where does validation happen?
-            IConfigurationSection connectionSection = _configuration.GetWebJobsConnectionStringSection(name);
+            IConfigurationSection connectionSection = configuration.GetWebJobsConnectionStringSection(name);
```

Comment:
This code should be more defensive. Even with the check that exists in `GetWebJobsConnectionStringSection` (actually indirectly in the method it ends up calling), not performing a `null` check here makes this a bit fragile. Please update to perform a check.